src/api/goods.ts

```
@@ -4,13 +4,26 @@ import { Good } from '../types/Good';
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
 export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+  return fetch(API_URL).then(response => {
+    if (!response.ok) {
+      // eslint-disable-next-line no-console
+      console.error('Failed to fetch users');
+
+      return [];
```

This violates the requirement that all API methods must propagate errors to the UI for proper error handling. Currently, if the response is not OK, you return an empty array instead of rejecting the promise. You should throw an error here so that the UI can catch it and display an error message as required by the checklist and description.